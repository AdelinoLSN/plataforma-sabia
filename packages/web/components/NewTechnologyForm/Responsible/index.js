/* eslint-disable no-use-before-define */
import React from 'react';
import PropTypes from 'prop-types';
import { useFieldArray, useForm, Controller } from 'react-hook-form';
import { InputField } from '../../Form';
import { Col, Row, Wrapper } from './styles';
import { CircularButton } from '../../Button';

const Responsible = ({ form }) => {
	const emptyValue = { fullName: '', email: '', phoneNumber: '', lattesID: '' };
	const initialValue = [emptyValue];

	const { control } = useForm({
		defaultValues: {
			responsible: initialValue,
		},
	});

	const { fields, append, remove } = useFieldArray({
		control,
		name: 'responsible',
	});

	const ResponsibleRow = (item, index) => {
		return (
			<Row key={index}>
				<Col size={2}>
					<InputField
						form={form}
						name={`responsible[${index}].fullName`}
						label="Nome Completo"
						placeholder="Nome do responsável"
						validation={{ required: true }}
					/>
				</Col>
				<Col>
					<InputField
						form={form}
						name={`responsible[${index}].email`}
						label="Email"
						placeholder="Ex.: email@dominio.com.br"
						validation={{ required: true }}
					/>
				</Col>
				<Col>
					<Controller
						as={
							<InputField
								form={form}
								name={`responsible[${index}].phone`}
								label="Telefone"
								placeholder="(xx) xxxx - xxxx"
								validation={{ required: true }}
							/>
						}
						name={`responsible[${index}].phone`}
						control={control}
						defaultValue={item.phone}
					/>
				</Col>
				<Col>
					<InputField
						form={form}
						name={`responsible[${index}].lattesId`}
						label="ID Lattes"
						placeholder="Somente números"
						validation={{ required: true }}
					/>
				</Col>

				<CircularButton
					disabled={index === 0 && fields.length === 1}
					small
					variant="remove"
					shortPadding
					onClick={(event) => {
						event.preventDefault();
						remove(index);
					}}
				>
					-
				</CircularButton>
			</Row>
		);
	};

	const mountInputFields = () => {
		return fields.map((element, index) => ResponsibleRow(element, index));
	};

	return (
		<Wrapper>
			{mountInputFields()}
			<CircularButton
				right
				variant="info"
				color="white"
				onClick={(event) => {
					event.preventDefault();
					append(emptyValue);
				}}
			>
				+
			</CircularButton>
		</Wrapper>
	);
};

Responsible.propTypes = {
	form: PropTypes.shape({}),
};

Responsible.defaultProps = {
	form: {},
};

export default Responsible;
