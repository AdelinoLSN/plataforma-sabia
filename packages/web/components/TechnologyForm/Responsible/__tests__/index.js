/* eslint-disable jest/no-disabled-tests */
import React from 'react';
import { render, fireEvent, screen } from 'test-utils';
import Responsible from '..';

jest.mock('react-icons/fa', () => {
	return {
		__esModule: true,
		FaPlus: () => {
			return <svg>FaPlus mocked</svg>;
		},
		FaMinus: () => {
			return <svg>FaMinus mocked</svg>;
		},
	};
});

test.skip('it renders the form with one row', () => {
	const { container } = render(
		<Responsible
			form={{
				errors: {},
				register: jest.fn(),
				control: {},
			}}
		/>,
	);

	expect(container).toMatchSnapshot();
});

test.skip('it increase the number of rows when clicking in add button', () => {
	render(
		<Responsible
			form={{
				errors: {},
				register: jest.fn(),
			}}
		/>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	expect(buttons[1]).toMatchInlineSnapshot(`
		.c0 {
		  background-color: hsla(195,86%,64%);
		  color: hsla(0,0%,100%);
		  border-radius: 100%;
		  height: 100%;
		  border: none;
		  font-size: 2rem;
		  text-transform: uppercase;
		  text-align: center;
		  -webkit-text-decoration: none;
		  text-decoration: none;
		  display: inline-block;
		  padding: 1rem !important;
		  float: right;
		  display: -webkit-box;
		  display: -webkit-flex;
		  display: -ms-flexbox;
		  display: flex;
		}

		.c0:hover {
		  opacity: 0.8;
		}

		@media (max-width:450px) {

		}

		@media (max-width:450px) {

		}

		@media (max-width:450px) {

		}

		<button
		  class="c0"
		  color="hsla(0, 0%, 100%)"
		  title=""
		  type="button"
		>
		  <svg>
		    FaPlus mocked
		  </svg>
		</button>
	`);
});

test.skip('it decrease the number of rows when clicking in remove button', () => {
	const { container } = render(
		<Responsible
			form={{
				errors: {},
				register: jest.fn(),
			}}
		/>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	const buttons2 = screen.getAllByRole('button');

	fireEvent.click(buttons2[0]);

	expect(container).toMatchSnapshot();
});

test.skip('it should not remove the row when only one is available', () => {
	const { container } = render(
		<Responsible
			form={{
				errors: {},
				register: jest.fn(),
			}}
		/>,
	);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[0]);

	expect(container).toMatchSnapshot();
});
