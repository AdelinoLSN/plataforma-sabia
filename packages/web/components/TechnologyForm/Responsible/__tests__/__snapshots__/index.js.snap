// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it decrease the number of rows when clicking in remove button 1`] = `
.c3 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c3 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c6::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c2 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].fullName"
          >
            Nome Completo
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Nome Completo"
              aria-required="true"
              class="c6"
              id="responsible[0].fullName"
              name="responsible[0].fullName"
              placeholder="Nome do responsável"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Email"
              aria-required="true"
              class="c6"
              id="responsible[0].email"
              name="responsible[0].email"
              placeholder="Ex.: email@dominio.com.br"
              type="text"
              value=""
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].phone"
          >
            Telefone
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Telefone"
              aria-required="true"
              class="c6"
              id="responsible[0].phone"
              name="responsible[0].phone"
              placeholder="(xx) xxxx - xxxx"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].lattesId"
          >
            ID Lattes
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="ID Lattes"
              aria-required="true"
              class="c6"
              id="responsible[0].lattesId"
              name="responsible[0].lattesId"
              placeholder="Somente números"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <button
        class="c9"
        color="hsla(0, 0%, 100%)"
        disabled=""
        title=""
        type="button"
      >
        <svg>
          FaMinus mocked
        </svg>
      </button>
    </div>
    <button
      class="c10"
      color="hsla(0, 0%, 100%)"
      title=""
      type="button"
    >
      <svg>
        FaPlus mocked
      </svg>
    </button>
  </div>
</div>
`;

exports[`it increase the number of rows when clicking in add button 1`] = `
.c3 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c3 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c6::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c2 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].fullName"
          >
            Nome Completo
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Nome Completo"
              aria-required="true"
              class="c6"
              id="responsible[0].fullName"
              name="responsible[0].fullName"
              placeholder="Nome do responsável"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Email"
              aria-required="true"
              class="c6"
              id="responsible[0].email"
              name="responsible[0].email"
              placeholder="Ex.: email@dominio.com.br"
              type="text"
              value=""
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].phone"
          >
            Telefone
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Telefone"
              aria-required="true"
              class="c6"
              id="responsible[0].phone"
              name="responsible[0].phone"
              placeholder="(xx) xxxx - xxxx"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].lattesId"
          >
            ID Lattes
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="ID Lattes"
              aria-required="true"
              class="c6"
              id="responsible[0].lattesId"
              name="responsible[0].lattesId"
              placeholder="Somente números"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <button
        class="c9"
        color="hsla(0, 0%, 100%)"
        title=""
        type="button"
      >
        <svg>
          FaMinus mocked
        </svg>
      </button>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
        size="2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[1].fullName"
          >
            Nome Completo
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Nome Completo"
              aria-required="true"
              class="c6"
              id="responsible[1].fullName"
              name="responsible[1].fullName"
              placeholder="Nome do responsável"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[1].email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Email"
              aria-required="true"
              class="c6"
              id="responsible[1].email"
              name="responsible[1].email"
              placeholder="Ex.: email@dominio.com.br"
              type="text"
              value=""
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[1].phone"
          >
            Telefone
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Telefone"
              aria-required="true"
              class="c6"
              id="responsible[1].phone"
              name="responsible[1].phone"
              placeholder="(xx) xxxx - xxxx"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[1].lattesId"
          >
            ID Lattes
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="ID Lattes"
              aria-required="true"
              class="c6"
              id="responsible[1].lattesId"
              name="responsible[1].lattesId"
              placeholder="Somente números"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <button
        class="c9"
        color="hsla(0, 0%, 100%)"
        title=""
        type="button"
      >
        <svg>
          FaMinus mocked
        </svg>
      </button>
    </div>
    <button
      class="c10"
      color="hsla(0, 0%, 100%)"
      title=""
      type="button"
    >
      <svg>
        FaPlus mocked
      </svg>
    </button>
  </div>
</div>
`;

exports[`it renders the form with one row 1`] = `
.c3 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c3 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c6::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c2 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].fullName"
          >
            Nome Completo
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Nome Completo"
              aria-required="true"
              class="c6"
              id="responsible[0].fullName"
              name="responsible[0].fullName"
              placeholder="Nome do responsável"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Email"
              aria-required="true"
              class="c6"
              id="responsible[0].email"
              name="responsible[0].email"
              placeholder="Ex.: email@dominio.com.br"
              type="text"
              value=""
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].phone"
          >
            Telefone
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Telefone"
              aria-required="true"
              class="c6"
              id="responsible[0].phone"
              name="responsible[0].phone"
              placeholder="(xx) xxxx - xxxx"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].lattesId"
          >
            ID Lattes
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="ID Lattes"
              aria-required="true"
              class="c6"
              id="responsible[0].lattesId"
              name="responsible[0].lattesId"
              placeholder="Somente números"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <button
        class="c9"
        color="hsla(0, 0%, 100%)"
        disabled=""
        title=""
        type="button"
      >
        <svg>
          FaMinus mocked
        </svg>
      </button>
    </div>
    <button
      class="c10"
      color="hsla(0, 0%, 100%)"
      title=""
      type="button"
    >
      <svg>
        FaPlus mocked
      </svg>
    </button>
  </div>
</div>
`;

exports[`it should not remove the row when only one is available 1`] = `
.c3 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c3 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c6::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 1rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

@media (max-width:450px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c2 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].fullName"
          >
            Nome Completo
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Nome Completo"
              aria-required="true"
              class="c6"
              id="responsible[0].fullName"
              name="responsible[0].fullName"
              placeholder="Nome do responsável"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Email"
              aria-required="true"
              class="c6"
              id="responsible[0].email"
              name="responsible[0].email"
              placeholder="Ex.: email@dominio.com.br"
              type="text"
              value=""
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].phone"
          >
            Telefone
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="Telefone"
              aria-required="true"
              class="c6"
              id="responsible[0].phone"
              name="responsible[0].phone"
              placeholder="(xx) xxxx - xxxx"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="responsible[0].lattesId"
          >
            ID Lattes
          </label>
          <div
            class="c5"
          >
            <input
              aria-label="ID Lattes"
              aria-required="true"
              class="c6"
              id="responsible[0].lattesId"
              name="responsible[0].lattesId"
              placeholder="Somente números"
              type="text"
            />
          </div>
          <span
            class="c7"
          />
        </div>
      </div>
      <button
        class="c9"
        color="hsla(0, 0%, 100%)"
        disabled=""
        title=""
        type="button"
      >
        <svg>
          FaMinus mocked
        </svg>
      </button>
    </div>
    <button
      class="c10"
      color="hsla(0, 0%, 100%)"
      title=""
      type="button"
    >
      <svg>
        FaPlus mocked
      </svg>
    </button>
  </div>
</div>
`;
